Feature/1 merge blueprint versions

ppxl opened the pull request:
Resolve #1 and implement a merged dogu/k8s command that is able to mirror mixed v1 and v2 blueprints

ppxl committed:
#1 Add Blueprint v1 structs from cesapp

This commit also fixes some minor typos, adds doc comments
and moves some higher-level definitions into bpcore since
v2 will probably make use of those as well.

ppxl committed:
#1 Add Blueprint mask v1 structs from cesapp

ppxl committed:
#1 export TargetStateIgnore

ppxl committed:
#1 Add blueprint v2 from blueprint-operator

ppxl committed:
#1 Exclude vendor and tmp directories

cesmarvin commented on v2/config.go:
```diff
@@ -0,0 +1,89 @@
+)
+
+// censorValue is the value for censoring sensitive blueprint configuration data.
+const censorValue = "*****"
```
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>const `censorValue` is unused
<!-- __reviewdog__:ChA2ZmQ4NzEzYjc5MjVmMmFiEgpjaGVja3N0eWxl -->

cesmarvin commented on v2/config.go:
```diff
@@ -0,0 +1,89 @@
+}
+
+// censorValues censors all sensitive configuration data to make them unrecognisable.
+func (config Config) censorValues() Config {
```
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `Config.censorValues` is unused
<!-- __reviewdog__:ChBjZWY2MzY0MDFkNjQxMGJiEgpjaGVja3N0eWxl -->

ppxl committed:
Revert "#1 Add blueprint v2 from blueprint-operator"

This reverts commit 060654f8b4eb530a1a0173b14ca11857d2b9a6b0.

ppxl committed:
#1 Import B.P. operator v2 code and make it compile

This import was meant to compile as-is but some changes were necessary
to make it compile without domain or kubernetes package references. Long
story short, I needed to change a lot of things

cesmarvin commented on bpcore/core.go:
```diff
@@ -0,0 +1,105 @@
+	TargetStateAbsent
+	// targetStateIgnore is currently only internally used to mark items that are present in the CES instance at hand
+	// but not mentioned in the blueprint.
+	targetStateIgnore
```
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>const `targetStateIgnore` is unused
<!-- __reviewdog__:ChA2MWNjMTVjOWQyN2VkYTFhEgpjaGVja3N0eWxl -->

ppxl committed:
#1 Export targetState "ignored"

ppxl committed:
#1 Change dogu mask target state to string type

This was changed during the extraction and was most likely a mistake.
The original code used string here.

ppxl committed:
#1 Change dogu target state to string type, change target state type

This was changed during the extraction and was most likely a mistake.
The original code used string here.

ppxl committed:
#1 Change component target state to string type

This was changed during the extraction and was most likely a mistake.
The original code used string here.

ppxl committed:
#1 Re-add missing deep-copy functions that went lost

ppxl committed:
#1 Implement missing deep-copy functions for TargetConfig

Somehow these functions seem to be necessary by the blueprint operator's
deep copy generator even though I do not see where I might have deleted
it

ppxl committed:
#1 Fix tests where implicit target should not be rendered

ppxl committed:
#1 Rename mis-named package v1 to json

In package names v1/v2/v${nonzerointeger} is reserved to identify module
versions. This was not the case of the previous state but rather the
version of the blueprint implementation.

This commit removes this misnaming towards json to amend the situation
above and also to put the focus on the format being worked on.

ppxl committed:
#1 Delete Blueprint v2

The kubernetes implementation Blueprint v2 was moved into its own
library k8s-blueprint-lib because there would be even more changes with
way more k8s resources and logic attached. These two functionalities
should not mix different libraries and possibly attach to another's core
entities.

ppxl committed:
#1 Fix project name after copy-pasta

ppxl committed:
#1 Simplify file layout and merge functions in existing files

ppxl committed:
#1 Add blueprint v1 parsing from cesapp v7

kahoona77 reviewed (approved):
LGTM
